kmcos/io.py

```diff
@@ -2865,7 +2866,8 @@ def _write_optimal_iftree_otf(self, items, indent, out):
                 out.write('%scall del_proc(%s, %s)\n' % (' ' * indent, item[1][0], rel_site))
 
         # and only keep those that have conditions
-        items = list(filter(lambda x: len(x[0]) > 1, items)) #like [x for x in items if x[0]]
+        # items = list(filter(lambda x: len(x[0]) > 1, items)) #like [x for x in items if x[0]]
+        items = [x for x in items if x[0]]
```

Comment:
The simpler syntax is probably better.  I made those comments but I am not actually sure if they are the same in all circumstances, which is why I made the comment rather than actually making the change.  Did you change to the simpler syntax because it fixes something? Or just for checking purposes? If it's just for checking purpose, I'd prefer to keep the more complicated syntax and the comment.

Reply:
Martin Comment: The items list (line 2868) did not result in the expected list. The 
proclist.f90 therefore did not contain add_proc calls (only del_proc), 
so there was nothing but zeros in the avail_sites (so no process to be 
executed, which gave the "deadlock" in the interval_search_real).